### DISCUSSION TITLE
How to add new boards

### DISCUSSION THREAD

**Discussion by gpeyton** [BIPES/BIPES#104] [Q&A] [answered]
I'm working on a custom ESP32-S2 board and would like to add it to BIPES. Is there documentation for how to do that? 

Thank you!

**Comment by gastmaier** [comment, marked as answer]
Hello!

For BIPES 2 (current release), just include the board in the [ ui/devinfo/devinfo.json](https://github.com/BIPES/BIPES/blob/master/ui/devinfo/devinfo.json) and some pinout image in the [ui/devinfo/media](https://github.com/BIPES/BIPES/tree/master/ui/devinfo/media)
You will notice that the  JSON has a `toolbox` parameter which links to the [toolboxes xmls](https://github.com/BIPES/BIPES/tree/master/ui/toolbox).
I guess the ESP32-S2  could use the already existing [esp32S2.xml](https://github.com/BIPES/BIPES/blob/master/ui/toolbox/esp32S2.xml) toolbox.

You might notice that the toolboxes xmls are unnecessarily verbose, since most blocks are shared between many devices; I would like to share that we are working hard on these kinds of issues for the next release  :).

Best regards,

**Reply by gpeyton** [reply]
Thanks @gastmaier! When do you expect to push out the next release?

**Reply by gastmaier** [reply]
Around March 1st I will release it.

**Comment by rafaelaroca** [comment]
Hello! @gastmaier covered all, but just to add 1 cent of information: as Jorge said, we have a working setup for ESP32-S2 using CircuitPython, which may be useful for what you need. There are some working examples using the Brazilian open source project Franzininho Wifi (a board based on the ESP32-S2) here: https://rafaelaroca.wordpress.com/2022/01/09/esp32s2-circuitpython-and-bipes-franzininho-wifi/.

regards.
Rafael
